src/env.cc

```diff
@@ -1562,14 +1562,31 @@ size_t Environment::NearHeapLimitCallback(void* data,
   size_t num_heap_spaces = env->isolate()->NumberOfHeapSpaces();
   for (size_t i = 0; i < num_heap_spaces; ++i) {
     env->isolate()->GetHeapSpaceStatistics(&stats, i);
+
+    Debug(env,
+          DebugCategory::DIAGNOSTICS,
+          "%s space_size = %" PRIu64 ", "
+          "space_used_size = %" PRIu64 ", "
+          "space_available_size = %" PRIu64 ", "
+          "physical_space_size = %" PRIu64 "\n",
+          stats.space_name(),
+          static_cast<uint64_t>(stats.space_size()),
+          static_cast<uint64_t>(stats.space_used_size()),
+          static_cast<uint64_t>(stats.space_available_size()),
+          static_cast<uint64_t>(stats.physical_space_size()));
+
+    // space_size() returns the allocated size of a given space,
+    // we use this to calculate the new limit because V8 also
+    // uses the allocated size to determine whether it should crash.
```

Comment:
```suggestion
    // space_size() returns the allocated size of a given space.
    // We use this to calculate the new limit because V8 also
    // uses the allocated size to determine whether it should crash.
```